### ISSUE TITLE
Beacon SDK adds 4.4 MB (30% of total) to app binary size

### ISSUE THREAD

**Issue by andreyz** [helpscout/beacon-ios-sdk#151]
- Beacon install method: 
    [ ] Carthage, 
    [ ] CocoaPods, 
    [x] Swift Package Manager,
    [ ] Manual
- SDK version: 2.2.0
- iOS OS version: 14.4
- Device make/model:
- Impacted user count:

**Actual outcome**

    Beacon adds 4.4 MB to app binary size, which is 30% of total app's binary size.

**Expected outcome**

    Beacon should take all steps possible to reduce it's binary size. Next largest third-party SDK in my app is 614.4 KB.

**Steps to reproduce**

    1. Follow steps from Create the App Size Report section of [Reducing Your App’s Size](https://developer.apple.com/documentation/xcode/reducing_your_app_s_size)
    2. Follow steps from Identify and Remove Unused Assets of [Doing Basic Optimization to Reduce Your App’s Size](https://developer.apple.com/documentation/xcode/reducing_your_app_s_size/doing_basic_optimization_to_reduce_your_app_s_size)
    3. Observe that Beacon.framework takes about 4.4 MB which is 30% of total binary size for my app's binary.

**Screenshots/Screen Recording**

<img width="276" alt="image" src="https://user-images.githubusercontent.com/205317/106257946-83d0b980-621d-11eb-8d3f-c74f77195281.png">

**Additional Information**

Please see referenced [Reducing Your App’s Size](https://developer.apple.com/documentation/xcode/reducing_your_app_s_size) and [Doing Basic Optimization to Reduce Your App’s Size](https://developer.apple.com/documentation/xcode/reducing_your_app_s_size/doing_basic_optimization_to_reduce_your_app_s_size) for suggestions on how to reduce the size.

**Comment by codenameDuchess** [comment]
Hi @andreyz 👋 

Thanks for bringing this issue to our attention and sorry for the trouble. We've opened an [internal tracking ticket](https://helpscout.atlassian.net/browse/IOSSDK-473) to investigate this issue and we'll be sure to reach back out once we've had a chance to address it.

**Comment by ScottPetit** [comment]
Hey @andreyz,

Just wanted to reach out and let you know that we addressed as much as we could with the 2.2.3 release.  We tried a few things related to code optimization (optimizing for size) that didn't really generate much of an effect.  We did have some assets that were still using 1x, 2x, 3x PNGs that we were able to transfer over to vector assets, some color assets that were duplicated, and we were able to get rid of the icon font to shave a bit off as well.  All in all I think we shaved ~250 KB off.  Thanks for the report.
